templates/todo/common/infra/bicep/app/apim-api.bicep

```diff
@@ -95,6 +98,22 @@ resource apimService 'Microsoft.ApiManagement/service@2021-08-01' existing = {
   name: name
 }
 
+// Necessary due to https://github.com/Azure/bicep/issues/9594
+var appNameForBicep = !empty(apiAppName) ? apiAppName : 'placeholderName'
+
+resource apiAppProperties 'Microsoft.Web/sites/config@2022-03-01' = if (!empty(apiAppName)) {
+  name: '${appNameForBicep}/web'
+  kind: 'string'
+  properties: {
+      apiManagementConfig: {
+        id: '${apimService.id}/apis/${apiName}'
+      }
+  }
+  dependsOn: [
+    apimService
+  ]
```

Comment:
Since you're directly referencing the `apimService` in the properties I don't think you will need the explicit `dependsOn` here since it should automatically be inferred by the dependent resource usage.

Reply:
Removed it while I was making other changes